### PR TITLE
docs: add APM landing page & update arch diagram

### PR DESCRIPTION
**Pull request by bmorelli25**
* Adds an APM landing page -- requested by users who were confused as to why there was no APM content in this guide
* Updates the architecture diagram to include synthetic monitoring and user experience

### CODE REVIEW

File: docs/en/observability/apm.asciidoc [hunk 1]
```diff
@@ -0,0 +1,12 @@
+[[apm]]
+= Application performance monitoring (APM)
+
+Elastic APM is an application performance monitoring system built on the Elastic Stack.
+It allows you to monitor software services and applications in real-time, by
```
Comment (dedemorton) [inline docs/en/observability/apm.asciidoc:5]:
```suggestion
It allows you to monitor software services and applications in real time, by
```

Comment (dedemorton) [reply docs/en/observability/apm.asciidoc:5]: minor, but typically it's real time (noun) and real-time (adj).

**Review by dedemorton — APPROVED** [review]
Just a minor suggestion. Otherwise, LGTM, so I will approve. Up to you whether you take the suggestion.

File: docs/en/observability/apm.asciidoc (hunk 1 above)
Comment (dedemorton) [reply docs/en/observability/apm.asciidoc:5]: BTW, I opened an issue in the style repo because this is a common mistake.

Comment (bmorelli25) [reply docs/en/observability/apm.asciidoc:5]: TIL

**Pull request merged**